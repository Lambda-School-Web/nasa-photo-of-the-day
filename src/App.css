.App {
  text-align: center;
  width: 100%;
}

.container {
  margin: 0 auto;
  max-width: 800px;
}

h1 {
  font-family: "Russo One", sans-serif;
  font-size: 3.5rem;
  color: #0b3d91;
}

img {
  max-width: 800px;
  height: auto;
}

h2,
p {
  font-family: "Roboto Mono", monospace;
}

h2 {
  color: #484848;
  font-size: 2.5rem;
}
p.desc {
  color: #767676;
  font-size: 1.3rem;
}
.footer {
  font-size: 0.9rem;
  color: #767676;
  display: flex;
  justify-content: space-between;
}

.desc {
  text-align: left;
}
