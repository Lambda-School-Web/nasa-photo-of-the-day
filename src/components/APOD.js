import React from "react";
import DatePicker from "react-date-picker";

function APOD(props) {
  const youtubeEmbed = (url, title) => {
    return (
      <iframe
        title={title}
        width="560"
        height="315"
        src={url}
        frameBorder="0"
        allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
        allowFullScreen
      ></iframe>
    );
  };
  const renderContent = data => {
    const { copyright, explanation, media_type, title, url } = data;
    return (
      <>
        {media_type === "image" ? (
          <img src={url} alt={title} />
        ) : (
          youtubeEmbed(url, title)
        )}

        <h2>{title}</h2>
        <div className="desc">
          <p className="desc">{explanation}</p>
        </div>
        <div className="footer">
          <DatePicker
            format="MM-dd-y"
            value={new Date(props.date)}
            clearIcon={null}
            onChange={e => props.callback(e)}
            maxDate={new Date()}
            minDate={new Date("June 16, 1995")}
          />
          {copyright && media_type === "image" ? (
            <p className="copyright">Image Credit & Copyright: {copyright}</p>
          ) : null}
          {copyright && media_type === "video" ? (
            <p className="copyright">Video Credit & License: {copyright}</p>
          ) : null}
        </div>
      </>
    );
  };

  return (
    <div className="container">
      {props.data ? renderContent(props.data) : <p>Loading...</p>}
    </div>
  );
}

export default APOD;
