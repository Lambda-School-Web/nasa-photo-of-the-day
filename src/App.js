import React, { useState, useEffect } from "react";
import APOD from "./components/APOD";
import nasaAPOD from "./api/NasaAPOD";
import { NasaH1, AppDiv, DescP } from "./styles/components";

function App() {
  const [data, setData] = useState(null);
  const [date, setDate] = useState("2019-09-12");

  useEffect(() => {
    nasaAPOD
      .get("", {
        params: { date: date }
      })
      .then(res => {
        setData(res.data);
        console.log(res.data);
      })
      .catch(err => console.log(err));
  }, [date]);

  const handleDateChange = date => {
    function changeMonth(date) {
      const months = [
        "Jan",
        "Feb",
        "Mar",
        "Apr",
        "May",
        "Jun",
        "Jul",
        "Aug",
        "Sep",
        "Oct",
        "Nov",
        "Dec"
      ];
      let monthNum = months.indexOf(date.substring(5, 8)) + 1;
      return monthNum < 10 ? `0${monthNum}` : `${monthNum}`;
    }

    const regex = /(\w{3}) (\d{2}) (\d{4})/;
    const regex2 = /[A-Za-z]+/;

    let newDate = date
      .toString()
      .substring(4, 15)
      .replace(regex, `$3-$1-$2`);

    newDate = newDate.replace(regex2, changeMonth(newDate));

    setDate(newDate);
  };

  return (
    <AppDiv>
      <NasaH1>Nasa's Astronomy Picture of the Day</NasaH1>
      {data ? (
        <APOD data={data} date={new Date(date)} callback={handleDateChange} />
      ) : (
        <DescP>Loading...</DescP>
      )}
    </AppDiv>
  );
}

export default App;
