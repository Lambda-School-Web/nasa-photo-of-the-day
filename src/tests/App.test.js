import React from "react";
import "@testing-library/jest-dom/extend-expect";
import * as rtl from "@testing-library/react";
import App from "../App";
import APOD from "../components/APOD";
afterEach(rtl.cleanup);

const dataAPOD = {
  copyright: "Markus Bauer",
  date: "2019-09-12",
  explanation:
    "These cosmic dust clouds drift some 1,300 light-years away along the fertile starfields of the constellation Cepheus. The beautiful Iris Nebula, also known as NGC 7023, blossoms at the upper left. Not the only nebula in the sky to evoke the imagery of flowers, its pretty, symmetric form spans about 6 light-years. This nebula's dominant blue color is characteristic of the pervasive dust grains reflecting light from a nearby hot, bluish star. But darker, obscuring dust clouds cover most of the nearly 4 degree wide field of view. At the right is the LDN 1147/1158 complex of Lynds Dark Nebulae. Stars are forming there, still hidden within the dark cloud cores. A search through the sharp image can identify Herbig-Haro objects though, jets of shocked glowing gas emanating from recently formed stars.",
  hdurl:
    "https://apod.nasa.gov/apod/image/1909/IRISNebulaSurroundingsNGC7023.jpg",
  media_type: "image",
  service_version: "v1",
  title: "The Iris Nebula in a Field of Dust",
  url:
    "https://apod.nasa.gov/apod/image/1909/IRISNebulaSurroundingsNGC7023_1100.jpg"
};
describe("App component", () => {
  it("renders correctly", () => {
    const AppComponent = rtl.render(<App />);
    expect(AppComponent.baseElement).toMatchSnapshot();
  });
});

describe("APOD component", () => {
  it("renders correctly", () => {
    const APODcomponent = rtl.render(
      <APOD data={dataAPOD} date={new Date("2019-09-12")} callback={null} />
    );
    expect(APODcomponent.baseElement).toMatchSnapshot();
  });
});
