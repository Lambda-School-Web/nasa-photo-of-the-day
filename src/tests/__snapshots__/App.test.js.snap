// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<body>
  <div>
    <div
      class="sc-bdVaJa bryXLM"
    >
      <h1
        class="sc-ifAKCX hJRqby"
      >
        Nasa's Astronomy Picture of the Day
      </h1>
      <p
        class="sc-gzVnrw sc-htoDjs evfIfr"
      >
        Loading...
      </p>
    </div>
  </div>
</body>
`;
